lib/components/file_picker/file_picker.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:file_picker/file_picker.dart';
+import 'package:flutter/material.dart';
+import 'package:tarka_ui/components/button/button.dart';
+import 'package:tarka_ui/components/button/style.dart';
+
+class TUIFilePicker extends StatefulWidget {
+  final bool allowMultipleFiles;
+  final String title;
+  final AllowedFileType allowedFileType;
+  final List<String>? allowedFileExtensions;
+  final Function(List<PlatformFile>?) filesPicked;
+
+  const TUIFilePicker({
+    super.key,
+    required this.title,
+    this.allowMultipleFiles = false,
+    this.allowedFileType = AllowedFileType.any,
+    this.allowedFileExtensions,
+    required this.filesPicked,
+  });
+
+  @override
+  State<TUIFilePicker> createState() => _TUIFilePickerState();
+}
+
+class _TUIFilePickerState extends State<TUIFilePicker> {
+  List<PlatformFile>? pickedFiles = [];
```

Comment:
Why do we hold selected files in the state? Can we pass the value directly to `filesPicked` callback?